world.js

```diff
@@ -120,6 +120,12 @@ var createWorldCreator = function() {
                     }
                 });
             });
+            elevator.on("indicatorstate_change", function() {
+                var elevator = this;
+                if(elevator.isOnAFloor() && !elevator.isMoving && !elevator.isFull()) {
```

Comment:
With this addition, the expression `elevator.isOnAFloor() && !elevator.isMoving && !elevator.isFull()` appears twice in `world.js`. I think it should be refactored into a function `elevator.isSuitableForUserEntrance()`. Though that could be done after your pull request is accepted.